### DISCUSSION TITLE
Turn/Lane instructions

### DISCUSSION THREAD

**Discussion by NickSlash** [osmandapp/OsmAnd#16345] [Maps Related]
I'm trying to determine why I get a "turn left" instruction for what is a "through" instruction in the map data. (my understanding of lane/turn mapping is minimal though)

Start 52.22431N 0.08985E -> End 52.21921N 0.10016E (Huntingdon Road, Cambridge, UK)

![map](https://user-images.githubusercontent.com/3884126/215333461-6a9003d7-3c2b-4a34-89af-9030ea8c3d82.png)
The Black Circle is roughly where I receive the turn left instruction in this example.

**Comment by vshcherb** [comment]
It's probably a bug https://osmand.net/map?start=52.226919%2C0.084565&end=52.220623%2C0.097203&mode=car#19/52.22204/0.09470 - but also tricky situation there is no lane assigned to https://www.openstreetmap.org/way/434898540, so algorithm treats it as "C" direction even though "TL|C" is for the next junction and not current. I think it could be changed slightly, As I see "TL|C" starts at a different point and service is not allowed at all to turn right. So I guess if that will be restricted somehow the problem will be gone
